source/Detector/Scaler/AMDSScalerDetector.cpp

```diff
@@ -265,11 +269,79 @@ void AMDSScalerDetector::onTriggerDwellTimerTimeout()
 {
 	if(triggerDwellInterfaceDwellModeControl_->withinTolerance(0.0)){
 		triggerDwellTimer_->stop();
-		triggerDwellInterfaceStartControl_->move(0);
 	}
 	emit requestFlattenedData(triggerDwellInterfaceDwellTimeControl_->value());
 }
 
+void AMDSScalerDetector::startScalerDwellAcquisition()
+{
+	double seconds = triggerDwellInterfaceDwellTimeControl_->value();
+	if(seconds > 0 && seconds < 100) {
+		if (triggerDwellInterfaceDwellModeControl_->withinTolerance(0.0)) {
+			// set the mapping of value changed. NOTE: This should be in a more graceful way
+			waitingChannels_.clear();
+
+			QMap<quint8, int> channelToIndexMap;
```

Comment:
This section of codes are stolen from the `setFlattenedData()`. 

It will be optimized later